sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/ResourceParserTests.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Net;
+using OpenTelemetry.Resources;
+using Xunit;
+
+namespace Azure.Monitor.OpenTelemetry.Exporter.Tests
+{
+    public class ResourceParserTests
+    {
+        [Fact]
+        public void NullResource()
+        {
+            var resourceParser = new ResourceParser();
+            resourceParser.UpdateRoleNameAndInstance(null);
+
+            Assert.Null(resourceParser.RoleName);
+            Assert.Null(resourceParser.RoleInstance);
+        }
+
+        [Fact]
+        public void DefaultResource()
```

Comment:
maybe add a test to validate that we only parse RoleName once, and will not parse any updated resource name.

Reply:
Good call out, covered it.